### DISCUSSION TITLE
Do you know how to fix this issue?

### DISCUSSION THREAD

**Discussion by ghost** [msys2/MINGW-packages#17801] [Q&A] [answered]
Step to reproduce: clone from my `mingw64` branch on https://github.com/iahunq/COS/tree/mingw64 and try to build with `make` on MINGW64. It will fail with this error:

```
MINGW64/release/Any.d:1: *** multiple target patterns.  Stop.
make: *** [CosBase/include/cos/mak/common:44: CosBase] Error 2
```

Strangely enough it will not fail on the MSYS environment.

@Biswa96

**Comment by Biswa96** [comment]
I got the following error.

```
$ make
CosBase/include/cos/prologue:119: CosBase/include/cos/cfg/MINGW64: No such file or directory
make: *** No rule to make target 'CosBase/include/cos/cfg/MINGW64'.  Stop.
```

**Reply by ghost** [reply]
You have to clone from the `mingw64` branch: https://github.com/iahunq/COS/tree/mingw64

`git clone -b mingw64 https://github.com/iahunq/COS`

**Reply by Biswa96** [reply]
My apology, I have no idea why it failed. But the issue does not seem related to msys2 or mingw64.

**Reply by ghost** [reply, marked as answer]
@Biswa96 Open `COS/CosBase/MINGW64/release/Any.d` you will see the paths all begin with `/msys64/mingw64/`. I tried to run the command `cd /msys64` and it failed with `-bash: cd: /msys64: No such file or directory`. So it seems the paths are invalid. `/msys64` should be replaced with `/`. The rule to generate `.d` files is on `COS/CosBase/include/cos/mak/compile`. I have no idea why it generated invalid paths on MINGW64. It works as expected on MSYS.

Update: I know how to fix this problem. The `sed` command on `COS/CosBase/include/cos/mak/compile` needs the `-z` switch to work as expected on MINGW64.
